src/Api/Controllers/OrganizationSponsorshipsController.cs

```diff
@@ -70,11 +70,13 @@ public class OrganizationSponsorshipsController : Controller
         [SelfHosted(NotSelfHostedOnly = true)]
         public async Task CreateSponsorship(Guid sponsoringOrgId, [FromBody] OrganizationSponsorshipCreateRequestModel model)
         {
+            var sponsoringOrg = await _organizationRepository.GetByIdAsync(sponsoringOrgId);
+
             var sponsorship = await _createSponsorshipCommand.CreateSponsorshipAsync(
-                await _organizationRepository.GetByIdAsync(sponsoringOrgId),
+                sponsoringOrg,
                 await _organizationUserRepository.GetByOrganizationAsync(sponsoringOrgId, _currentContext.UserId ?? default),
                 model.PlanSponsorshipType, model.SponsoredEmail, model.FriendlyName);
-            await _sendSponsorshipOfferCommand.SendSponsorshipOfferAsync(sponsorship, (await CurrentUser).Email);
+            await _sendSponsorshipOfferCommand.SendSponsorshipOfferAsync(sponsorship, sponsoringOrg.Name);
```

Comment:
maaaan this is the danger of primitive types.

Reply:
Yeah...